### PR TITLE
Add Grid (table) component

### PR DESCRIPTION
**Pull request by Pavel0408**

### CODE REVIEW

File: packages/grid/package.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    "clean": "rimraf dist lib .publish"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com:xcritical-software/xc-front-kit.git"
+  },
+  "keywords": [],
+  "author": "XCritical Software",
+  "license": "MIT",
+  "homepage": "https://github.com/xcritical-software/xc-front-kit/tree/develop/packages/item#readme",
```
[the remark this answers is not in the document]
Comment (Pavel0408) [reply packages/grid/package.json]: done

File: packages/grid/src/grid/Grid.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+} from '../interfaces';
+import { Row } from '../row/Row';
+import {
+  ContentWrapper, HeadWrapper, TableHead, TableStyled, TableWrapper, HeaderStyled,
+} from '../styled/styled';
+import { gridTheme } from '../utils/get-styles';
+
+
+export const Grid: React.FC<ITable> = React.memo((props: ITableProps) => {
+  const { rows, columns, theme = {} } = props;
```
[the remark this answers is not in the document]
Comment (Pavel0408) [reply packages/grid/src/grid/Grid.tsx]: done

File: packages/grid/src/grid/Grid.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  const handleScroll = (): void => {
+    const scrolledDiv = tableWrapperElement.current;
+    if (scrolledDiv) {
+      setTableScroll(scrolledDiv.scrollLeft);
+    }
+  };
+
+  const handleSelectRows = useCallback((row: IRow): void => {
+    if (selectedRows.some((el) => el === row)) {
+      const newRows = [...selectedRows].filter((el) => row !== el);
```
[the remark this answers is not in the document]
Comment (Pavel0408) [reply packages/grid/src/grid/Grid.tsx]: done

File: packages/grid/src/row/Row.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+} from 'react';
+import PlusBoxOutlineIcon from 'mdi-react/PlusBoxOutlineIcon';
+import MinusBoxOutlineIcon from 'mdi-react/MinusBoxOutlineIcon';
+import {
+  StyledCell, RowStyled, ToggleButton, RowShift, TableDataStyled,
+} from '../styled/styled';
+import { IColumn, IRow, IRowData } from '../interfaces';
+
+
+export const Row: React.FC<IRow> = React.memo(({
```
[the remark this answers is not in the document]
Comment (Pavel0408) [reply packages/grid/src/row/Row.tsx]: done

File: packages/grid/package.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    "polished": "^3.0.0",
+    "prop-types": "^15.0.0",
+    "react": "^16.8.3",
+    "styled-components": "^4.1.0"
+  },
+  "dependencies": {
+    "@xcritical/theme": "^0.1.0",
+    "lodash.get": "4.x",
+    "mdi-react": "^6.2.0",
+    "polished": "^3.0.0"
```
Comment (ISp1rit) [inline packages/grid/package.json]: polished is unused dependency.

File: packages/grid/package.json [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  },
+  "keywords": [],
+  "author": "XCritical Software",
+  "license": "MIT",
+  "homepage": "https://github.com/xcritical-software/xc-front-kit/tree/develop/packages/grid#readme",
+  "private": false,
+  "publishConfig": { "access": "public" },
+  "peerDependencies": {
+    "polished": "^3.0.0",
+    "prop-types": "^15.0.0",
```
Comment (ISp1rit) [inline packages/grid/package.json]: prop-types is unused dependency.

File: packages/grid/src/Grid.tsx [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    themeRef.current = gridTheme<ITableTheme>(theme);
+  }, [theme]);
+
+  return (
+    <ContentWrapper theme={ themeRef.current }>
+      <HeadWrapper>
+        <TableHead style={ { transform: `translateX(-${tableScroll}px)` } } theme={ themeRef.current }>
+          { columns.map(({ title, width }: IColumn) => (
+            <HeaderStyled
+              // columnName={ title }
```
Comment (ISp1rit) [inline packages/grid/src/Grid.tsx]: Unnecessary comment?

File: packages/grid/package.json (hunk 5 above)
Comment (Pavel0408) [reply packages/grid/package.json]: done

File: packages/grid/package.json (hunk 6 above)
Comment (Pavel0408) [reply packages/grid/package.json]: done

File: packages/grid/src/Grid.tsx (hunk 7 above)
Comment (Pavel0408) [reply packages/grid/src/Grid.tsx]: done

**Pull request merged**